dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/SearchServiceBean.java

```diff
@@ -918,35 +925,101 @@ private String getPermissionFilterQuery(DataverseRequest dataverseRequest, SolrQ
 
     }
 
-    public String getLocaleFacetName(String name) {
-        return getLocaleFacetName(name, datasetFieldService.findAllOrderedByName());
+    public String getLocaledFacetName(String facetValue, String facetFieldGroupName) {
+        return getLocaledFacetName(facetValue, datasetFieldService.findAllOrderedByName(), facetFieldGroupName);
     }
 
-    public String getLocaleFacetName(String name, List<DatasetFieldType> datasetFields ) {
-        final String key = toBundleNameFormat(name);
-        try {
-            String displayName = getStringFromBundle(format("facets.search.fieldtype.%s.label", name));
-            if(isNotBlank(displayName)) {
-                return displayName;
-            }
-            displayName = BundleUtil.getStringFromBundle(name);
-            if(isNotBlank(displayName)) {
-                return displayName;
-            }
-            displayName = BundleUtil.getStringFromBundle("dataverse.type.selectTab." + key);
-            if(isNotBlank(displayName)) {
-                return displayName;
-            }
-            for(DatasetFieldType datasetField : datasetFields) {
-                displayName = BundleUtil.getStringFromPropertyFile("controlledvocabulary."
-                                + datasetField.getName() + "." + key,
-                        datasetField.getMetadataBlock().getName().toLowerCase());
-                if(isNotBlank(displayName)) {
-                    return displayName;
+    public String getLocaledFacetName(String facetFieldGroupName) {
+        return getLocaledFacetName(facetFieldGroupName, datasetFieldService.findAllOrderedByName(), facetFieldGroupName);
```

Comment:
Also passing the same value twice as different args to make some deduction afterwards is code smell.